### PR TITLE
Make versions consistent

### PR DESCRIPTION
**Pull request by niemela**
This PR should not contain anything controversial, I will merge it fairly quickly...

- Make front matter and newlines consistent between versions, to simplify diffing
- Copy changes between versions (so that that should be same is).
- Add explanation of the ICPC subset buttons
- Change some terminology:
  - `contest control system` to `judging system`
  - `standard` to `format`

### CODE REVIEW

File: spec/2023-07-draft.md [hunk 1]
```diff
@@ -35,13 +35,13 @@ All floating point numbers must be given as the external character sequences def
 ### Programs
 
 There are a number of different kinds of programs that may be provided in the problem package: submissions, input validators, and output validators.
-All programs are always represented by a single file or directory.
-In other words, if a program consists of several files, these must be provided in a single directory.
+All programs are always represented by a single file or directory.
```
Comment (RagnarGrootKoerkamp) [inline spec/2023-07-draft.md:42]: trailing spaces everywhere???

File: spec/2023-07-draft.md [hunk 2]
```diff
@@ -265,20 +265,20 @@ At the top level, the test data is divided into exactly two groups: `sample` and
 <div class="not-icpc">
 
 The <em>result</em> of a test data group is computed by applying a <em>grader</em> to all of the sub-results (test cases and subgroups) in the group. 
-See [Graders](#graders) for more details.
+See [Graders](#graders "wikilink") for more details.
```
Comment (RagnarGrootKoerkamp) [inline spec/2023-07-draft.md:272]: The 'wikilink' hovertext is useless.

File: spec/2023-07-draft.md [hunk 3]
```diff
@@ -430,7 +430,7 @@ The meaning of the parameters listed above are:
   The answer file may, but is not necessarily required to, contain the "correct answer" for the problem. 
   For example, it might contain the output which was produced by a judge's solution for the problem when run with input file as input. 
   Alternatively, the "answer file" might contain information, in arbitrary format, which instructs the validator in some way about how to accomplish its task. 
-  The meaning of the contents of the answer file is not defined by this standard.
+  The meaning of the contents of the answer file is not defined by this format.
```
Comment (RagnarGrootKoerkamp) [inline spec/2023-07-draft.md:433]: I think this is somewhat confusing, given the use of `format` in the sentence above. `standard` feels clearer to me in general, or remove 'by this format' completely.

**Review by RagnarGrootKoerkamp — COMMENTED** [review]
some small comments that can be fixed later

File: spec/2023-07-draft.md (hunk 3 above)
Comment (jsannemo) [reply spec/2023-07-draft.md:433]: In my head the text has always been a standard specifying a problem format (in the sense that RFCs are standards defining protocols).

**Pull request merged**